src/Core/ClassInfo.php

```diff
@@ -301,14 +301,14 @@ public static function classImplements($className, $interfaceName)
      */
     public static function classes_for_file($filePath)
     {
-        $absFilePath = Director::getAbsFile($filePath);
+        $absFilePath = Director::getAbsFile(realpath($filePath));
         $classManifest = ClassLoader::inst()->getManifest();
         $classes = $classManifest->getClasses();
         $classNames = $classManifest->getClassNames();
 
         $matchedClasses = [];
         foreach ($classes as $lowerClass => $compareFilePath) {
-            if (strcasecmp($absFilePath ?? '', $compareFilePath ?? '') === 0) {
+            if (strcasecmp($absFilePath ?? '', realpath($compareFilePath) ?? '') === 0) {
```

Comment:
Would this also work?

```suggestion
            if (strcasecmp($absFilePath ?? '', Director::getAbsFile($compareFilePath) ?? '') === 0) {
```

The change as it is indicates to me that `Director::getAbsFile` could return different result e.g. symlinks or hardlinks are used, is that the case? If so, we should move the realpath call into that method. If not, and the above suggestions also achieves the same result, I'd prefer that.

Reply:
@michalkleiner If I remove `realpath()` and log the output of the two variables I get this:

```
$absFilePath = D:\workspace\diasorintv\vendor\dnadesign\silverstripe-elemental-userforms\src/Model/ElementForm.php
$compareFilePath = D:\workspace\diasorintv/vendor/silverstripe/framework/src/Security/Member_GroupSet.php
```

So even though `$absFilePath` is run through `Director::getAbsFile()`, it is not normalised/cleaned up.

Do you want me to add `realpath()` to all return values in `Director::getAbsFile()` and use `Director::getAbsFile()` here for both paths?

Reply:
So all it needs is fixing the mixed up path delimiters? That's the issue?

Reply:
Yes.

Reply:
Ok, thanks for the confirmation. I'm sorry but this feels like an issue coming from somewhere else and it should be fixed at its source. Not even inside `getAbsFile`, but didn't have time to dig deeper. Here it feels like slapping a bandaid on it.

Reply:
I agree. But there are over 100 hard coded paths using `/` instead of `DIRECTORY_SEPARATOR` in framework alone. 

I don't really care where the band-aid goes. At the moment, `i18nTextCollector` doesn't work on Windows because of this. So if we can fix this here, then at least getting the translations works again. 

Fixing all the hard coded paths is a bigger task that I am not willing to take on.

Reply:
PHP is pretty lenient when it comes to path formats for normal operations. 
But here we string-compare two paths that were concatenated in different places, with different sets of paths and directory separators. That's why it fails here and not elsewhere.

Reply:
Can we use `Convert::slashes` for the paths then? I'd be ok with that for the comparison, assuming it may come in different/mixed formats from the underlying data. The `realpath` unpacks symbolic links and may actually change the path to something else.

Reply:
Sorry for the radio silence. Yes, `Convert::slashes` works too.